nni/algorithms/compression/pytorch/quantization/quantizers.py

```diff
@@ -391,3 +527,37 @@ def quantize_output(self, output, wrapper, **kwargs):
         # remove zeros
         out[out == 0] = 1
         return out
+
+    def export_model(self, model_path, calibration_path=None, onnx_path=None, input_shape=None, device=None):
+        """
+        Export quantized model weights and calibration parameters(optional)
+
+        Parameters
+        ----------
+        model_path : str
+            path to save quantized model weight
+        calibration_path : str
+            (optional) path to save quantize parameters after calibration
+        onnx_path : str
+            (optional) path to save onnx model
+        input_shape : list or tuple
+            input shape to onnx model
+        device : torch.device
+            device of the model, used to place the dummy input tensor for exporting onnx file.
+            the tensor is placed on cpu if ```device``` is None
+        """
+        assert model_path is not None, 'model_path must be specified'
+        self._unwrap_model()
+
+        torch.save(self.bound_model.state_dict(), model_path)
+        logger.info('Model state_dict saved to %s', model_path)
+        if calibration_path is not None:
+            logger.info('No calibration config will be saved because no calibration data in BNN quantizer')
+        if onnx_path is not None:
+            assert input_shape is not None, 'input_shape must be specified to export onnx model'
+            # input info needed
+            if device is None:
+                device = torch.device('cpu')
+            input_data = torch.Tensor(*input_shape)
+            torch.onnx.export(self.bound_model, input_data.to(device), onnx_path)
+            logger.info('Model in onnx with input shape %s saved to %s', input_data.shape, onnx_path)
```

Comment:
For other implementations return a dict, add `return {}` is better?

And all `export model()` implementations seem to have mostly the same logic, can we use the implementation in `QAT_Quantizer` for all `Quantizer`? Or just specify how to construct `calibration_config` in different `Quantizer`?

Reply:
Useful suggestions! Have modified.